components/infobox/extensions/commons/age_calculation.lua

```diff
@@ -0,0 +1,154 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:AgeCalculation
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local AgeCalculation = {}
+local Class = require('Module:Class')
+
+local _LANG = mw.getContentLanguage()
+
+local _EPOCH = _LANG:formatDate('c', '1970-1-1')
+local _EPOCH_FIELD = { 1970, 1, 1 }
+local _CURRENT_FIELD = os.date('*t', os.time())
+local _CURRENT_ISO = _LANG:formatDate('c')
+local _CURRENT_YEAR = tonumber(_LANG:formatDate('Y'))
+local _DAYS_IN_MONTH = { 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31 }
+
+function AgeCalculation.run(birth, birthLocation, death, personType, shouldStore)
+	--disect the dates into fields
+	local birthFields = mw.text.split(birth or '', '-')
+	local deathFields = mw.text.split(death or '', '-')
+	--strip them to numbers
+	for i = 1, 3 do
+		birthFields[i] = tonumber(birthFields[i] or '')
+		deathFields[i] = tonumber(deathFields[i] or '')
+	end
+
+	birthFields = AgeCalculation._processDateFields(birthFields)
+	deathFields = AgeCalculation._processDateFields(deathFields)
```

Comment:
Can be a function instead:

```lua
	local birthFields = AgeCalculation._parseDate(birth) -- This is now an array of dates
	local deathFields = AgeCalculation._parseDate(death)
```

Aside from the `_processDateFields` function, which I have comments on below. I would prefer starting like above, and then going from there.